src/process.c

```diff
@@ -111,7 +111,8 @@ static int luv_spawn(lua_State* L) {
         uv_stream_t* stream = luv_check_stream(L, -1);
         int err = uv_fileno((uv_handle_t*)stream, &fd);
         if (err == UV_EINVAL || err == UV_EBADF) {
-          options.stdio[i].flags = UV_CREATE_PIPE | UV_READABLE_PIPE | UV_WRITABLE_PIPE;
+          options.stdio[i].flags =
+              UV_CREATE_PIPE | (i == 0 ? UV_READABLE_PIPE : UV_WRITABLE_PIPE);
```

Comment:
This is great, but I'm unsure it's something we can assume.
We should probably use read/write for fds above 2 right? 
How are higher FDs typically used?  I've used them to pass TCP FDs from master load balancer to child worker processes.